library/Hasql/Transaction/Sessions.hs

```diff
@@ -9,11 +9,12 @@ where
 
 import Hasql.Session qualified as B
 import Hasql.Transaction.Config qualified as C
+import Hasql.Transaction.Private.Prelude
 import Hasql.Transaction.Private.Transaction qualified as A
 
 -- |
 -- Execute the transaction using the provided isolation level and mode.
 {-# INLINE transaction #-}
-transaction :: C.IsolationLevel -> C.Mode -> A.Transaction a -> B.Session a
-transaction isolation mode transaction =
-  A.run transaction isolation mode
+transaction :: C.IsolationLevel -> C.Mode -> Bool -> A.Transaction a -> B.Session a
+transaction isolation mode retryOnError transaction =
+  A.run transaction isolation mode retryOnError
```

Comment:
If we want to avoid breaking changes, I can add another function here, something like `transactionNoRetry`.

Reply:
Good idea!

Reply:
Done!